deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -339,11 +362,20 @@ func addCandidatesForNewChainLogic(e deployment.Environment, c AddCandidatesForN
 				SkipChainConfigValidation: true,
 			},
 		},
+		// use currentDonId here
 		DonIDOverrides: map[uint64]uint32{c.NewChain.Selector: donID},
 	})
 	if err != nil {
 		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SetCandidateChangeset on home chain: %w", err)
 	}
+
+	// Claim donID using donIDClaimer at the end of the changeset run
+	txOpts := e.Chains[c.HomeChainSelector].DeployerKey
+	tx, err := state.Chains[c.HomeChainSelector].DonIDClaimer.ClaimNextDONId(txOpts)
```

Comment:
Let's have a check whether deployer key is authorizedDeployer

Reply:
Q - if the changeset fails to execute would this claimed don Id be just left behind ?

Reply:
we are claimingDonID as a final operation for this changeset, so if it fails here we basically have to re-run the entire changeset which donID won't be claimed